LibOS/shim/src/fs/shim_dcache.c

```diff
@@ -170,9 +177,15 @@ static inline void __dput_dentry (struct shim_dentry * dent)
         /* move the node to unused list unless it is persistent */
         if (!(dent->state & DENTRY_PERSIST)) {
             dent->state |= DENTRY_RECENTLY;
-            list_del(&dent->list);
-            INIT_LIST_HEAD(&dent->list);
-            list_add(&dent->list, &unused);
+            /* DEP: The only reference I see to this field is being placed on 
+             * the unused list. It seems that sometimes the dent is not
+             * on a list; just hack this for now; this code will be reworked
+             * soon anyway.
+             */
+
+            /* Chia-Che 10/17/17: Originally we place all the freed
+             * dentries on an unused list, now we just maintain a long list
+             * of all dentries. */
```

Comment:
@donporter Please check this.

Instead of maintaining a free list, we will just maintain a long dentry list. (In the future, we will walk this list to free unused dentries.)